libs/api/domains/air-discount-scheme/src/lib/flight-leg/flight-leg.resolver.ts

```diff
@@ -0,0 +1,38 @@
+import { UseGuards } from '@nestjs/common'
+import { Args, Parent, Query, ResolveField, Resolver } from '@nestjs/graphql'
+
+import { ApiScope } from '@island.is/auth/scopes'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import type { FlightLeg as TFlightLeg } from '@island.is/clients/air-discount-scheme'
+
+import { FlightLegsInput } from './dto/flight-leg.input'
+import { FlightLeg } from '../models/flightLeg.model'
+import { FlightLegService } from './flight-leg.service'
+import { Audit } from '@island.is/nest/audit'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Audit({ namespace: '@island.is/air-discount-scheme' })
+@Resolver(() => FlightLeg)
+export class FlightLegResolver {
```

Comment:
Shouldn't this be in the flight-leg-admin folder and require the admin scope, since it uses the admin api?

Reply:
Yeah, or a different approach entirely.
I want to populate flightlegs of a user and I should be using a resolveField for that instead of writing a whole new resolver, bit of a brainfart there.

I want that lookup for a public facing user.

On facing away from the admin api: 
I'm having a problem with assigning an endpoint multiple tags because the generated api export file has errors stemming from something like
```typescript
export * from ./adminApi
export * from ./publicApi
```
will result in a type error upon exporting the same endpoint.

I'll close this for now and come back with a different approach but all insights are welcome.